src/LayoutManager/LayoutManager.ts

```diff
@@ -130,11 +142,27 @@ export class LayoutManager {
     const { target } = context;
     const { canvas } = target;
     // handle layout triggers subscription
+    // @TODO: gate the registration when the group is interactive
     if (
       context.type === LAYOUT_TYPE_INITIALIZATION ||
       context.type === LAYOUT_TYPE_ADDED
     ) {
-      this.subscribeTargets(context);
+      const withDifferentParent = context.targets.filter((object) =>
+        this.subscribe(object, context)
+      );
+      // if objects have a different parent from group, they are in an active seleciton
+      // so we need to subscribe the active selection event to trigger the parent performLayout
+      withDifferentParent.forEach(({ group, parent }) => {
+        if (parent) {
+          // we may subscribe an active selection more than once,
```

Comment:
this doesn't sound good, bad for perf and for logic

Reply:
This is in contrast with wanting to register all the objects all the time.
Right now if an active selection has 2 objects in the same group i think will try to register twice, but trying a different approach my fix that.